src/Entities/Factory.php

```diff
@@ -6,9 +6,11 @@ abstract class Factory
 {
     abstract public static function make(array $data, string $bot_username): Entity;
 
-    public static function resolveEntityClass(string $class, array $property, string $bot_username = ''): Entity
+    public static function resolveEntityClass(string $class, mixed $property, string $bot_username = ''): Entity
     {
-        if (is_subclass_of($class, Factory::class)) {
+        if (is_a($property, $class)) {
+            return $property;
+        } elseif (is_subclass_of($class, Factory::class)) {
```

Comment:
I added this check, so we can use subEntities() on classes that don't come from Telegram like in KeyboardButton. So if the user decides to set an object property with an array like this `setWebApp(['url' => 'https://example.com'])` it will still gets parsed into a WebAppInfo object if the user calls `getWebApp()` later on...

Reply:
Fancy!

Tiny change to have cleaner returns:
```suggestion
        }
        
        if (is_subclass_of($class, Factory::class)) {
```